crates/bevy_render/src/render_resource/pipeline_cache.rs

```diff
@@ -468,14 +468,14 @@ impl LayoutCache {
 /// pipeline object is deferred to the [`RenderSet::Render`] step, just before the render
 /// graph starts being processed, as this requires access to the GPU.
 ///
-/// Note that the cache do not perform automatic deduplication of identical pipelines. It is
+/// Note that the cache does not perform automatic deduplication of identical pipelines. It is
 /// up to the user not to insert the same pipeline twice to avoid wasting GPU resources.
 ///
 /// [`RenderSet::Render`]: crate::RenderSet::Render
 #[derive(Resource)]
 pub struct PipelineCache {
-    layout_cache: LayoutCache,
-    shader_cache: ShaderCache,
+    layout_cache: Arc<Mutex<LayoutCache>>,
+    shader_cache: Arc<Mutex<ShaderCache>>,
```

Comment:
It looks like these are `Send + Sync` and read-heavy. Would it be worth evaluating the performance of `parking_lot::RwLock` here?

Reply:
Agreed

Reply:
These are not read-heavy. We only ever lock them for writing.